### PR TITLE
[Fix] Don’t check filenames on text input.

### LINKED ISSUE #662: jsx-filename-extension errors on "executeOnText"
When using the JS API for eslint, I'm getting jsx-filename-extension errors.

```js
import { CLIEngine } from 'eslint';
const cli = new CLIEngine({
  useEslintrc: false,
  plugins: ['react'],
  parserOptions: {
    ecmaFeatures: {
      jsx: true,
    },
  },
  ecmaFeatures: {
    jsx: true
  },
  rules: {
    'react/jsx-filename-extension': 2,
  },
});

const { results } = cli.executeOnText(`
var React = require('react');
module.exports = function SFC() { return <div />; };
`);
```
I would expect `results[0].messages` to be an empty array, since `results[0].filePath` is `<text>`.

However, I get:
```js
{ filePath: '<text>',
  messages: 
   [ { ruleId: 'react/jsx-filename-extension',
       severity: 2,
       message: 'JSX not allowed in files with extension \'\'',
       line: 3,
       column: 42,
       nodeType: 'JSXElement',
       source: 'module.exports = function SFC() { return <div />; };' } ],
  errorCount: 1,
  warningCount: 0 }
```

### PR DESCRIPTION
**Pull request by ljharb**
Fixes #662.

### CODE REVIEW

File: lib/rules/jsx-filename-extension.js [hunk 1]
```diff
@@ -32,9 +32,12 @@ module.exports = function(context) {
   return {
 
     JSXElement: function(node) {
-      var allowedExtensions = getExtensionsConfig();
       var filename = context.getFilename();
+      if (filename === '<text>') {
```
Comment (lencioni) [inline lib/rules/jsx-filename-extension.js]: I wonder if this comparison could reference a variable or something instead of hard-coding `<text>`

**Comment by lencioni** [comment]
LGTM! I didn't even realize this was a thing.

File: lib/rules/jsx-filename-extension.js (hunk 1 above)
Comment (ljharb) [inline lib/rules/jsx-filename-extension.js]: Sadly the only place it exists is https://github.com/eslint/eslint/blob/master/lib/cli-engine.js#L220

**Comment by ljharb** [comment]
@yannickcr ping :-)

**Comment by yannickcr** [comment]
Merged, thanks!

**Pull request merged**